sdk.yaml

```diff
@@ -248,6 +248,13 @@ Realtime:
       .synopsis: |
         The option to configure a `cipher`, specifying encryption algorithm and attributes.
         Used for subscribe and publish on this channel.
+    Mode:
+      .documentation:
+        - https://faqs.ably.com/what-are-channel-mode-flags-and-how-do-i-use-them
+      .specification: [RTL16, TB2d, TR3, RTS3c]
+      .synopsis: |
+        Specify a subset of the permissions given in your token/key for use with this channel, specified using channel mode flags.
+        This feature uses Ably service-side filtering, so it can be used to save unnecessary messages counting towards your usage such as preventing presence updates going to all users.
```

Comment:
I know this is from the FAQ, but I don't quite understand what it means. If a user doesn't want a client to, say, enter presence, well then why just not have that client enter presence, or just not give them the capability? I don't think it's very clear from this description what these modes are for.

Reply:
@SimonWoolf or @paddybyers could you assist here, as I don't understand this feature enough either.

Reply:
> why just not have that client enter presence
That works fine for presence. But what if i want to attach to a channel to enter presence, but I'm never going to subscribe to any messages? Modes gives you a way to say 'I don't care about receiving messages, please don't send me any'. Or 'I only care about normal messages not presence messages' (which means it's a useful way round the n^2 problem -- clients that enter presence but don't need to see presence don't have to subscribe to it)

Differs from capabilities in 2 ways:
- it's optional and specified by the client itself, vs capabilities which are mandatory and specified by the key or customer's auth server
- it's more granualar - eg we have a `subscribe` capability, but that doesn't distinguish between message and presence subscriptions, modes can.

(..the latter was arguably an API design mistake and we should have had a 1:1 mapping. But, well, this is where we are now.)

Reply:
Thanks, both. I've included @SimonWoolf's commentary in https://github.com/ably/features/pull/33/commits/5e6e46206967557327c8e18ab8b66d358d62327b

I'm now going to resolve this conversation on @lmars' behalf so that I can get this pull request merged. We can always further iterate this synopsis later on, in a subsequent PR...

Reply:
discussion issues for context:
- prehistory / ideation: https://github.com/ably/ideas/issues/3
- API discussion: https://github.com/ably/docs/issues/279
- Final spec PR: https://github.com/ably/docs/pull/766